dvc/repo/metrics/show.py

```diff
@@ -102,33 +99,21 @@ def show(
     recursive=False,
     revs=None,
     all_commits=False,
+    onerror=None,
 ):
-    res = {}
-    metrics_found = False
+    if onerror is None:
+        onerror = collect_error
 
+    res = {}
     for rev in repo.brancher(
         revs=revs,
         all_branches=all_branches,
         all_tags=all_tags,
         all_commits=all_commits,
     ):
-        metrics = _collect_metrics(repo, targets, rev, recursive)
-
-        if not metrics_found and metrics:
-            metrics_found = True
-
-        vals = _read_metrics(repo, metrics, rev)
-
-        if vals:
-            res[rev] = vals
-
-    if not res:
-        if metrics_found:
-            raise NoMetricsParsedError("metrics")
-        elif targets:
-            raise MetricDoesNotExistError(targets)
-        else:
-            raise NoMetricsFoundError("metrics", "-m/-M")
+        res[rev] = error_handler(_gather_metrics)(
+            repo, targets, rev, recursive, onerror=onerror
```

Comment:
It would probably be easier to read if we would pass `onerror` directly to `error_handler`, but since we need that down the line (to handle errors araising during particular file parsing) I decided to left it in `kwargs`

Reply:
Passing the `onerror` seems a little bit clumsy, maybe we should revisit context flags change? 
https://github.com/iterative/dvc/pull/2957